src/encoder.rs

```diff
@@ -159,12 +161,22 @@ impl<InstructionsType: Instructions> Encoder<InstructionsType> {
         };
     }
 
+    fn push_path_condition(&mut self, cond: SMTVariable) {
+        self.path_conditions.push(cond);
+    }
+
+    fn pop_path_condition(&mut self) {
+        self.path_conditions.pop();
```

Comment:
```suggestion
        self.path_conditions.pop().unwrap();
```
to assert it is not empty?

Reply:
pop will already panic on that I guess

Reply:
ah true